File: Documents/Projects/litcode/interview/IAS/binarySearch.py
def binary_search(arr, l, r, t):

    if l <= r:
        m = int((l+r)/2)

        if t == arr[m]:
            return m
        elif t < arr[m]:
            return binary_search(arr, l, m-1, t)
        elif t > arr[m]:
            return binary_search(arr, m+1, r, t)

    return -1

if __name__ == '__main__':

    arr = [0,1,2,3,4,5,6,7,8,9]
    t = 1
    print(binary_search(arr, 0, len(arr)-1, t))

File: Documents/Projects/litcode/interview/IAS/binarySearch.java
class binarySearch{

    public static int binSearch(int[] arr, int l, int r, int target){
        if(l <= r){

            int m = (l + r)/2;

            if(arr[m] == target){
                return m;
            }
            if(target > arr[m]){
                return binSearch(arr, m+1, r, target);
            }
            if(target < arr[m]){
                return binSearch(arr, l, m-1, target);
            }
        }

        return -1;
    }


    public static void main(String[] args){
        int[] arr = {1,5,7,9,10,12,52,55,67,90};

        int target = 52;
        int location = binSearch(arr, 0, arr.length-1, target);

        System.out.println(location);

    }
}